liblocation/src/main/java/com/mapbox/android/core/location/LocationEngineProxy.java

```diff
@@ -31,7 +31,11 @@ public void requestLocationUpdates(@NonNull LocationEngineRequest request,
                                      @Nullable Looper looper) throws SecurityException {
     checkNotNull(request, "request == null");
     checkNotNull(callback, "callback == null");
-    locationEngineImpl.requestLocationUpdates(request, getListener(callback),
+    T listener = getListener(callback);
+    if (listener instanceof LocationCallbackTransport) {
```

Comment:
so, why'd we cast every `listener` of every location request when we'd need this only for `AndroidLocationEngine` and its children?

Reply:
I see, you mean we handle this logic in `AndroidLocationEngineImp`?